src/components/main-view/main-view.jsx

```diff
@@ -40,18 +56,23 @@ class MainView extends React.Component {
       });
   }
 
-  componentDidMount() {
-    let accessToken = localStorage.getItem('token');
-    if (accessToken !== null) {
-      this.setState({
-        user: localStorage.getItem('user')
+  getUserData(token) {
```

Comment:
this is very clever and I see how you are utilizing this to pass userData onto your profile update component. We are going to tweak it just a little bit more to make this easier for ourselves.